lib/src/components/NxFormSelect/__tests__/NxFormSelect.test.tsx

```diff
@@ -5,45 +5,490 @@
+          }
+
+          it('has no validation alert and no select a11y error message', function() {
+            const component = quickRender();
+
+            expect(component.queryByRole('alert')).not.toBeInTheDocument();
+            expect(component.getByRole('combobox')).not.toHaveErrorMessage();
+          });
+
+          it('does not set aria-invalid on the select', function() {
+            expect(quickRender().getByRole('combobox')).not.toHaveAttribute('aria-invalid', 'true');
+          });
+        });
+      });
+
+      describe('when there are validation errors', function() {
+        const singleValidationErrorsMinimalProps = { ...pristineMinimalProps, validationErrors: 'foo' },
+            multiValidationErrorsMinimalProps = { ...pristineMinimalProps, validationErrors: ['bar', 'foo'] },
+            singleRender = rtlRender(NxFormSelect, singleValidationErrorsMinimalProps),
+            multiRender = rtlRender(NxFormSelect, multiValidationErrorsMinimalProps);
+
+        it('has no validation alert and no select a11y error message', function() {
+          const singleError = singleRender(),
+              multiError = multiRender();
+
+          expect(singleError.queryByRole('alert')).not.toBeInTheDocument();
+          expect(singleError.getByRole('combobox')).not.toHaveErrorMessage();
+
+          expect(multiError.queryByRole('alert')).not.toBeInTheDocument();
+          expect(multiError.getByRole('combobox')).not.toHaveErrorMessage();
+        });
+
+        it('does not set aria-invalid on the select', function() {
+          expect(singleRender().getByRole('combobox')).not.toHaveAttribute('aria-invalid', 'true');
+          expect(multiRender().getByRole('combobox')).not.toHaveAttribute('aria-invalid', 'true');
+        });
+
+        describe('when in a form when showValidationErrors', function() {
+          function quickRender(extraProps?: Partial<Props>) {
+            const renderResult = render(
+              <NxForm showValidationErrors onSubmit={() => {}}>
+                <NxFormSelect data-testid="select" { ...nonValidatableMinimalProps } { ...extraProps } />
+              </NxForm>
+            );
+
+            const boundQueries = within(renderResult.container);
+
+            return { ...renderResult, ...boundQueries };
+          }
+
+          it('has no validation alert and no select a11y error message', function() {
+            const component = quickRender();
+
+            expect(component.queryByRole('alert')).not.toBeInTheDocument();
+            expect(component.getByRole('combobox')).not.toHaveErrorMessage();
+          });
+
+          it('does not set aria-invalid on the select', function() {
+            expect(quickRender().getByRole('combobox')).not.toHaveAttribute('aria-invalid', 'true');
+          });
+        });
+      });
+    });
+
+    describe('when not pristine', function() {
+      const pristineMinimalProps = { ...nonValidatableMinimalProps, isPristine: true };
```

Comment:
if it's not pristine, should isPristine be set to false?

Reply:
> if it's not pristine, should isPristine be set to false?

Ah yes.  It wasn't failing because 1. this is the "when not validatable" section, and 2. the "when not validatable" props were also wrong. 28ef4ec78